cmd/containerd/main.go

```diff
@@ -188,17 +168,46 @@ func serveMetrics(ctx gocontext.Context, address string) {
 	}
 }
 
-func serveGRPC(ctx gocontext.Context, server *grpc.Server, l net.Listener) {
-	defer l.Close()
-	if err := server.Serve(l); err != nil {
-		log.G(ctx).WithError(err).Fatal("GRPC server failure")
+func serveGRPC(ctx gocontext.Context, context *cli.Context, server *grpc.Server) error {
+	path := context.GlobalString("socket")
```

Comment:
nit: since you're only need the `cli.Context` for this, just pass the socket pas as a parameter